geth/notification/fcm/notifier.go

```diff
@@ -0,0 +1,55 @@
+package fcm
+
+import (
+	"github.com/NaySoftware/go-fcm"
+	"github.com/status-im/status-go/geth/common"
+	"github.com/status-im/status-go/geth/notification"
+)
+
+// Notifier represents messaging provider for notifications.
+type Notifier struct {
+	client firebaseClient
+}
+
+// NewNotifier Firebase Cloud Messaging client constructor.
+func NewNotifier(key string) func() common.Notifier {
```

Comment:
I suppose the return type in the function signature should be `common.NotifierConstructor`? Technically they're almost the same but this option is better for understanding.

Also, I'm still trying to come up with proper naming. Now that you've made it a constructor (which is definitely a good idea!), it's not `Notifier` any more. It's `Notification` and should have a single `Send` method instead of `Notify`.
Here https://github.com/status-im/status-go/pull/390/files#diff-ebd449640936a8f327a6a4a1c3274d7cR32 I'd also named it `newNotification` so that the usage is `api.b.newNotification()`. By the way, can't we just create a separate constructor for `Notification` and just call it before sending a notification? It'll be a way simpler approach.

What do you think of all that?

Reply:
Done.